### PR TITLE
Canvas.py doc enhancement

### PR DESCRIPTION
**Pull request by Nusnus**

### CODE REVIEW

File: celery/canvas.py [hunk 1]
```diff
@@ -1046,6 +1062,38 @@ def prepare_steps(self, args, kwargs, tasks,
...
+        To execute a chain, we first need to unpack it correctly.
+        During the unpacking, we might encounter other chains, groups, or chords,
+        which we need to unpack as well.
+
+        For example:
+        chain(signature1, chain(signature2, signature3)) --> Upgrades to chain(signature1, signature2, signature3)
+        chain(group(signature1, signature2), signature3) --> Upgrades to chord([signature1, signature2], signature3)
+
+        The responsibility of this method is to assure that the chain is
+        correctly unpacked, and that the correct callbacks are set up along the way.
```
Comment (auvipy) [inline celery/canvas.py:1076]:
```suggestion
        correctly unpacked, and then the correct callbacks are set up along the way.
```

File: celery/canvas.py [hunk 2]
```diff
@@ -1046,6 +1062,38 @@ def prepare_steps(self, args, kwargs, tasks,
                       last_task_id=None, group_id=None, chord_body=None,
                       clone=True, from_dict=Signature.from_dict,
                       group_index=None):
+        """Prepare the chain for execution.
+
+        To execute a chain, we first need to unpack it correctly.
+        During the unpacking, we might encounter other chains, groups, or chords,
```
Comment (auvipy) [inline celery/canvas.py:1068]:
```suggestion
        During the unpacking, we might encounter other chains, groups, or chords
```

File: celery/canvas.py [hunk 3]
```diff
@@ -1046,6 +1062,38 @@ def prepare_steps(self, args, kwargs, tasks,
...
+        which we need to unpack as well.
+
+        For example:
+        chain(signature1, chain(signature2, signature3)) --> Upgrades to chain(signature1, signature2, signature3)
+        chain(group(signature1, signature2), signature3) --> Upgrades to chord([signature1, signature2], signature3)
+
+        The responsibility of this method is to assure that the chain is
```
Comment (auvipy) [inline celery/canvas.py:1075]:
```suggestion
        The responsibility of this method is to ensure that the chain is
```

**Review by auvipy — CHANGES_REQUESTED** [review]

**Comment by Nusnus** [comment]
Thanks for the review @auvipy
I've applied your suggestions, good catch!

**Review by auvipy — APPROVED** [review]

**Pull request merged**